frontend/src/lib/components/topbar/PullRequest.svelte

```diff
@@ -0,0 +1,662 @@
+<!-- PullRequest.svelte -->
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import { ToastType, addToast } from '$lib/toast';
+	import {
+		apiAddress,
+		branchName,
+		currentFile,
+		me,
+		openIssues,
+		selectedIssues,
+		openPullRequests,
+		baseBranch
+	} from '$lib/main';
+	import type { Issue } from '$lib/types';
+	import { get } from 'svelte/store';
+	import LoadingIcon from '../elements/LoadingIcon.svelte';
+
+	let showModal = false;
+	let prCommit = '';
+	let isExpanded: { [key: number]: boolean } = {};
+	let showLoadingIcon: boolean;
+	let selectedPullRequest: number | null = null;
+	let prAuthor = '';
+
+	function openModal() {
+		showModal = true;
+		checkOpenPullRequests();
+		document.addEventListener('keydown', handleEscape);
+	}
+
+	function closeModal() {
+		selectedIssues.set([]);
+		prCommit = '';
+		showModal = false;
+		selectedPullRequest = null;
+		showLoadingIcon = false;
+
+		document.removeEventListener('keydown', handleEscape);
+	}
+
+	function toggleExpand(issueId: number) {
+		isExpanded[issueId] = !isExpanded[issueId];
+	}
+
+	function handleEscape(event: KeyboardEvent) {
+		if (event.key === 'Escape') {
+			closeModal();
+		}
+	}
+
+	function toggleSelection(issue: Issue) {
+		selectedIssues.update((issues: Issue[]) => {
+			const idx = issues.findIndex((i) => i.id === issue.id);
+			if (idx !== -1) {
+				issues.splice(idx, 1);
+			} else {
+				issues.push(issue);
+			}
+			return [...issues];
+		});
+	}
+
+	onMount(() => {
+		getOpenIssues();
+	});
+
+	async function getOpenIssues() {
+		const state = 'open';
+		const labels = '';
+		const url = `${apiAddress}/api/issues/${state}${labels ? `?labels=${labels}` : ''}`;
+
+		try {
+			// Fetch the data from the API
+			const response = await fetch(url, {
+				method: 'GET',
+				credentials: 'include'
+			});
+
+			// Check if the response is successful (status code 2xx)
+			if (!response.ok) {
+				const errorMessage = `Failed to fetch open issues. (Code ${response.status}: "${response.statusText}")`;
+				addToast({
+					message: errorMessage,
+					type: ToastType.Error,
+					dismissible: true
+				});
+				return;
+			}
+
+			// Parse the response as JSON
+			const responseData = await response.json();
+
+			// Validate the response structure
+			if (responseData.status === 'success' && Array.isArray(responseData.data?.issues)) {
+				const issuesOnly = responseData.data.issues.filter((issue: Issue) => !issue.pull_request);
+				const pullRequestsOnly = responseData.data.issues.filter(
+					(issue: Issue) => issue.pull_request
+				);
+				openIssues.set(issuesOnly);
+				openPullRequests.set(pullRequestsOnly);
+			} else {
+				// Handle unexpected response structure
+				const errorMessage = `Unexpected response structure: ${JSON.stringify(responseData)}`;
+				addToast({
+					message: errorMessage,
+					type: ToastType.Error,
+					dismissible: true
+				});
+			}
+		} catch (error: unknown) {
```

Comment:
Just wipe this `try/catch` entirely, I want fatal errors to break shit so people report bugs